src/Type/Php/ArraySliceFunctionReturnTypeExtension.php

```diff
@@ -62,7 +62,7 @@ public function getTypeFromFunctionCall(
 			$limit = new NullType();
 		}
 
-		$constantArrays = TypeUtils::getConstantArrays($valueType);
```

Comment:
could you give a small explanation what the semantic difference of these 2 get*Array methods is?
If seen you used it in some places while fixing stuff in past PRs, but by reading the code itself I cannot make sense of it.

Reply:
Well obviously I thought that `TypeUtils::getConstantArrays` would be the way to go forward and the old version would go away, but the old version is more efficient.

The difference is that if you have `array{0: 'foo', 1?: 'bar'}`, `TypeUtils::getConstantArrays` calls `ConstantArrayType::getArrays()` and that returns a union of two arrays:

* `array{'foo'}`
* `array{'foo', 'bar'}`

So code that uses `TypeUtils::getConstantArrays` doesn't need to care about optional keys. The problem is that with multiple optional keys, this explodes in many combinations and can become slow. So what I'm doing in this PR is to use `TypeUtils::getOldConstantArrays`, have less ConstantArrayTypes in a union and handle optional keys at each place explicitly.